dialogs/simulation/reference_density_dialog.py

```diff
@@ -97,14 +99,41 @@ def __init__(self, recoil_element: RecoilElement,
         self.formLayout.removeRow(self.widget)
 
         self.isOk = False
+        self.isCancelled = False  # Cancel-button's state
 
         self.exec_()
 
     def _state_changed(self):
         if self.userSelectionCheckBox.isChecked():
-            self.scientific_spinbox.setDisabled(False)
+            # When the reference density is set by manually and the user
```

Comment:
> The cancel button doesn't work right. If I click Set the reference density manually, Ok (critical dialog) and then Cancel (reference density dialog), the checkbox stays enabled, but the numerical value doesn't persist. The yellow warning label doesn't show up either.

A good catch. I tried to be sharp.

> I would remove the critical dialog completely, and just draw a red warning text in the _Reference density_ dialog itself.

Why?

Reply:
> > I would remove the critical dialog completely, and just draw a red warning text in the _Reference density_ dialog itself.
> 
> Why?

Because with another dialog that has *Ok* and *Cancel* buttons, there's more stuff to keep track of. If the new dialog were removed, or at least it had its *Cancel* button removed, the code would get simpler.